src/deadline_worker_agent/sessions/attachment_models.py

```diff
@@ -0,0 +1,175 @@
+        Convert to an OpenJD-compatible path mapping rule.
+        Returns:
+            PathMappingRule: A path mapping rule using local root path as destination
+        Raises:
+            ValueError: If local_root_path is empty
+        """
+        if not self.local_root_path:
+            raise ValueError("local_root_path must be set to create path mapping rule")
+
+        return PathMappingRule(
+            source_path_format=self.manifest_properties.rootPathFormat.value,
+            source_path=self.manifest_properties.rootPath,
+            destination_path=self.local_root_path,
+        )
+
+    def get_hashed_source_path(self) -> str:
+        """
+        Generate a hashed string from file system location name and root path.
+        Returns:
+            str: A hash string generated from the combination of file_system_location_name
+                 and root_path using the manifest's default hash algorithm
+        Note:
+            This follows the same pattern used in asset_sync.py for manifest name generation.
+        """
+        hash_alg = AssetManifest.get_default_hash_alg()
+        return hash_data(
+            f"{self.file_system_location_name or ''}{self.root_path}".encode(), hash_alg
+        )
+
+    @classmethod
+    def from_manifest_properties(
+        cls,
+        manifest_properties: ManifestProperties,
+        local_root_path: str,
+        local_manifest_paths: Optional[List[str]] = None,
+    ) -> "WorkerManifestProperties":
+        """
+        Create WorkerManifestProperties from ManifestProperties and local paths.
+        Args:
+            manifest_properties: The original manifest properties
+            local_root_path: Local root path for attachment files
+            local_manifest_paths: Optional list of local paths for manifest files (supports step dependencies)
+        Returns:
+            WorkerManifestProperties: A new worker manifest properties instance
+        """
+        return cls(
+            manifest_properties=manifest_properties,
+            local_manifest_paths=local_manifest_paths or [],
+            local_root_path=local_root_path,
+        )
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Convert WorkerManifestProperties to a dictionary for JSON serialization.
+        Returns:
+            Dict[str, Any]: A dictionary representation of the worker manifest properties
+        """
+        return {
+            "manifest_properties": {
+                "root_path": self.manifest_properties.rootPath,
+                "root_path_format": self.manifest_properties.rootPathFormat.value,
+                "file_system_location_name": self.manifest_properties.fileSystemLocationName,
+                "input_manifest_path": self.manifest_properties.inputManifestPath,
+                "input_manifest_hash": self.manifest_properties.inputManifestHash,
+                "output_relative_directories": self.manifest_properties.outputRelativeDirectories,
+            },
```

Comment:
The `ManifestProperties` class has a `to_dict` member (see https://github.com/aws-deadline/deadline-cloud/blob/67869c6ba09fce023fd37d0ee5e9981f1f4b83eb/src/deadline/job_attachments/models.py#L206). This could call that member to shorten the code and remove the duplication.

In `ManifestProperties.to_dict`, the dict/json form of these objects is using camel case instead of snake case, it would be nice to follow that convention in this `to_dict` as well.

```suggestion
            "manifestProperties": self.manifest_properties.to_dict(),
```

Reply:
Thanks for the feedback. Let me explain my reasoning.

Regarding the naming convention, `ManifestProperties` currently uses camelCase because it's used for parsing API responses. Since there's no specific requirement for camelCase in this context, I opted for snake_case to maintain Python conventions.

I agree with your suggestion about using ManifestProperties.to_dict. The maintainability and consistency with the model outweigh the Python convention benefits. Future API changes would only need updates in one place, which reduces risk of de-synchronization between the model and its usage

I will update to use this approach.